custom_components/sat/cycles.py

```
@@ -0,0 +1,532 @@
+    @property
+    def started_since(self) -> Optional[float]:
+        return self._current_cycle_start
+
+    def reset(self) -> None:
+        """Reset the internal tracking state (does not clear history)."""
+        self._current_samples.clear()
+        self._last_flame_active = None
+        self._current_cycle_start = None
+        self._last_flame_off_timestamp = None
+
+    def update(self, boiler_state: BoilerState, pwm_status: PWMStatus, timestamp: float = None) -> None:
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)